lib/bolt/applicator.rb

```diff
@@ -235,9 +226,23 @@ def apply_ast(raw_ast, targets, options, plan_vars = {})
                   result
                 end
               else
+
+                plugin_tarball = Concurrent::Delay.new do
```

Comment:
Will this be built once for each target because it's inside the `queue_execute` method?

Reply:
Yes it was. I have changed it so it is calculated before the target loop.